### ISSUE TITLE
Changes needed to BNN example for compatibility with tensorflow 2.0-alpha?

### ISSUE THREAD

**Issue by bjfar** [tensorflow/probability#438]
I am interested in trying out the Bayesian neural network example at https://github.com/tensorflow/probability/blob/master/tensorflow_probability/examples/bayesian_neural_network.py, however I have TensorFlow 2.0-alpha installed, because the latest versions of tensorflow-probability require it. I am looking for a bit of helping making the necessary changes to get it working. The first error I encountered was this:

```
  File "BNN_tensorflow_example.py", line 44, in <module>
    from tensorflow.contrib.learn.python.learn.datasets import mnist
ModuleNotFoundError: No module named 'tensorflow.contrib'
```

So ok I guess the datasets just moved. I tried getting them this way instead:

```
import tensorflow_datasets as tfds
datasets, info = tfds.load(name='mnist', with_info=True, as_supervised=True)
mnist = datasets['train']
```

And that seems to work. At least I see the network being trained, according to the output messages. However at some point the following error occurs:

```
Traceback (most recent call last):
  File "BNN_tensorflow_example.py", line 346, in <module>
    tf.compat.v1.app.run()
  File "/home/farmer/anaconda3/envs/general/lib/python3.6/site-packages/tensorflow/python/platform/app.py", line 40, in run
    _run(main=main, argv=argv, flags_parser=_parse_flags_tolerate_undef)
  File "/home/farmer/anaconda3/envs/general/lib/python3.6/site-packages/absl/app.py", line 300, in run
    _run_main(main, args)
  File "/home/farmer/anaconda3/envs/general/lib/python3.6/site-packages/absl/app.py", line 251, in _run_main
    sys.exit(main(argv))
  File "BNN_tensorflow_example.py", line 227, in main
    mnist_data = mnist.read_data_sets(FLAGS.data_dir, reshape=False)
AttributeError: '_OptionsDataset' object has no attribute 'read_data_sets'
```

This one looks tougher. Any ideas? Perhaps something changed in the dataset API?

**Comment by brianwa84** [comment]
Our examples have not yet been ported to TF2. The library itself should be
compatible with tf-nightly and the TF2 nightly.
There is also a TF 1.14 RC and a TF 2 alpha RC you could try, which should
both be compatible with our 0.7 RC.

If you want to rewrite the example and make it TF2 friendly, we'd love a PR.

Brian Patton | Software Engineer | bjp@google.com

**Comment by nbro** [comment]
@bjfar You can use Keras to load the MNIST dataset. See https://keras.io/datasets/. It should no be too difficult to do it, even if you're not familiar with Keras. I will eventually provide a pull request to the Bayesian neural network example, with some updates, so that it is compatible with TF 2.

**Comment by srvasude** [comment]
I think this has been fixed by #698 . Closing this, but feel free to reopen if you see additional problems with TF2.
